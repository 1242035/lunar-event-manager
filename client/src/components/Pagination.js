import React from 'react';

/**
 * Implements pagination for switching pages.
 */
class Pagination extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activePageLink: 1,
    };
  }

  changePage = (e) => {
    const pageNumber = Number(e.currentTarget.dataset.id);
    this.props.getPage(pageNumber);
    this.setState({ activePageLink: pageNumber });
  }

  checkActive = (pageNumber) => {
    if (this.state.activePageLink === pageNumber) return 'active';
    return '';
  }

  render() {
    return (
      <div className="io-col-center">
        <ul className="pagination">
          <li id="pagePrevious" className="page-item disabled"><span className="page-link">&lt;&lt;</span></li>
          <li className={`page-item ${this.checkActive(1)}`}><span className="page-link" data-id="1" onClick={this.changePage}>1</span></li>
          <li className={`page-item ${this.checkActive(2)}`}><span className="page-link" data-id="2" onClick={this.changePage}>2</span></li>
          <li className={`page-item ${this.checkActive(3)}`}><span className="page-link" data-id="3" onClick={this.changePage}>3</span></li>
          <li className={`page-item ${this.checkActive(4)}`}><span className="page-link" data-id="4" onClick={this.changePage}>4</span></li>
          <li id="pageNext" className="page-item disabled"><span className="page-link">&gt;&gt;</span></li>
        </ul>
      </div>
    );
  }
}

export default Pagination;
