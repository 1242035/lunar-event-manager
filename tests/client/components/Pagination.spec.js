import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../../../client/src/components/Pagination';

describe('Pagination', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render component', () => {
    wrapper = shallow(<Pagination />);
    expect(wrapper.length).toEqual(1);
  });
});
