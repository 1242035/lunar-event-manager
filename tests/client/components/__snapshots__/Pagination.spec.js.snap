// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="pagination"
      >
        <li
          className="page-item disabled"
          id="pagePrevious"
        >
          <span
            className="page-link"
          >
            &lt;&lt;
          </span>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            data-id="1"
            onClick={[Function]}
          >
            1
          </span>
        </li>
        <li
          className="page-item "
        >
          <span
            className="page-link"
            data-id="2"
            onClick={[Function]}
          >
            2
          </span>
        </li>
        <li
          className="page-item "
        >
          <span
            className="page-link"
            data-id="3"
            onClick={[Function]}
          >
            3
          </span>
        </li>
        <li
          className="page-item "
        >
          <span
            className="page-link"
            data-id="4"
            onClick={[Function]}
          >
            4
          </span>
        </li>
        <li
          className="page-item disabled"
          id="pageNext"
        >
          <span
            className="page-link"
          >
            &gt;&gt;
          </span>
        </li>
      </ul>,
      "className": "io-col-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="page-item disabled"
            id="pagePrevious"
          >
            <span
              className="page-link"
            >
              &lt;&lt;
            </span>
          </li>,
          <li
            className="page-item active"
          >
            <span
              className="page-link"
              data-id="1"
              onClick={[Function]}
            >
              1
            </span>
          </li>,
          <li
            className="page-item "
          >
            <span
              className="page-link"
              data-id="2"
              onClick={[Function]}
            >
              2
            </span>
          </li>,
          <li
            className="page-item "
          >
            <span
              className="page-link"
              data-id="3"
              onClick={[Function]}
            >
              3
            </span>
          </li>,
          <li
            className="page-item "
          >
            <span
              className="page-link"
              data-id="4"
              onClick={[Function]}
            >
              4
            </span>
          </li>,
          <li
            className="page-item disabled"
            id="pageNext"
          >
            <span
              className="page-link"
            >
              &gt;&gt;
            </span>
          </li>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="page-link"
            >
              &lt;&lt;
            </span>,
            "className": "page-item disabled",
            "id": "pagePrevious",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "<<",
              "className": "page-link",
            },
            "ref": null,
            "rendered": "<<",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="page-link"
              data-id="1"
              onClick={[Function]}
            >
              1
            </span>,
            "className": "page-item active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
              "className": "page-link",
              "data-id": "1",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "1",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="page-link"
              data-id="2"
              onClick={[Function]}
            >
              2
            </span>,
            "className": "page-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2",
              "className": "page-link",
              "data-id": "2",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "2",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="page-link"
              data-id="3"
              onClick={[Function]}
            >
              3
            </span>,
            "className": "page-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3",
              "className": "page-link",
              "data-id": "3",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "3",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="page-link"
              data-id="4"
              onClick={[Function]}
            >
              4
            </span>,
            "className": "page-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4",
              "className": "page-link",
              "data-id": "4",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "4",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="page-link"
            >
              &gt;&gt;
            </span>,
            "className": "page-item disabled",
            "id": "pageNext",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": ">>",
              "className": "page-link",
            },
            "ref": null,
            "rendered": ">>",
            "type": "span",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
            id="pagePrevious"
          >
            <span
              className="page-link"
            >
              &lt;&lt;
            </span>
          </li>
          <li
            className="page-item active"
          >
            <span
              className="page-link"
              data-id="1"
              onClick={[Function]}
            >
              1
            </span>
          </li>
          <li
            className="page-item "
          >
            <span
              className="page-link"
              data-id="2"
              onClick={[Function]}
            >
              2
            </span>
          </li>
          <li
            className="page-item "
          >
            <span
              className="page-link"
              data-id="3"
              onClick={[Function]}
            >
              3
            </span>
          </li>
          <li
            className="page-item "
          >
            <span
              className="page-link"
              data-id="4"
              onClick={[Function]}
            >
              4
            </span>
          </li>
          <li
            className="page-item disabled"
            id="pageNext"
          >
            <span
              className="page-link"
            >
              &gt;&gt;
            </span>
          </li>
        </ul>,
        "className": "io-col-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="page-item disabled"
              id="pagePrevious"
            >
              <span
                className="page-link"
              >
                &lt;&lt;
              </span>
            </li>,
            <li
              className="page-item active"
            >
              <span
                className="page-link"
                data-id="1"
                onClick={[Function]}
              >
                1
              </span>
            </li>,
            <li
              className="page-item "
            >
              <span
                className="page-link"
                data-id="2"
                onClick={[Function]}
              >
                2
              </span>
            </li>,
            <li
              className="page-item "
            >
              <span
                className="page-link"
                data-id="3"
                onClick={[Function]}
              >
                3
              </span>
            </li>,
            <li
              className="page-item "
            >
              <span
                className="page-link"
                data-id="4"
                onClick={[Function]}
              >
                4
              </span>
            </li>,
            <li
              className="page-item disabled"
              id="pageNext"
            >
              <span
                className="page-link"
              >
                &gt;&gt;
              </span>
            </li>,
          ],
          "className": "pagination",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="page-link"
              >
                &lt;&lt;
              </span>,
              "className": "page-item disabled",
              "id": "pagePrevious",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "<<",
                "className": "page-link",
              },
              "ref": null,
              "rendered": "<<",
              "type": "span",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="page-link"
                data-id="1"
                onClick={[Function]}
              >
                1
              </span>,
              "className": "page-item active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
                "className": "page-link",
                "data-id": "1",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "1",
              "type": "span",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="page-link"
                data-id="2"
                onClick={[Function]}
              >
                2
              </span>,
              "className": "page-item ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2",
                "className": "page-link",
                "data-id": "2",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "2",
              "type": "span",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="page-link"
                data-id="3"
                onClick={[Function]}
              >
                3
              </span>,
              "className": "page-item ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3",
                "className": "page-link",
                "data-id": "3",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "3",
              "type": "span",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="page-link"
                data-id="4"
                onClick={[Function]}
              >
                4
              </span>,
              "className": "page-item ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4",
                "className": "page-link",
                "data-id": "4",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "4",
              "type": "span",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="page-link"
              >
                &gt;&gt;
              </span>,
              "className": "page-item disabled",
              "id": "pageNext",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": ">>",
                "className": "page-link",
              },
              "ref": null,
              "rendered": ">>",
              "type": "span",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
